src/config.cr

```diff
@@ -7,7 +7,7 @@ module Shards
 
   VERSION_REFERENCE     = /^v?\d+[-.][-.a-zA-Z\d]+$/
   VERSION_TAG           = /^v(\d+[-.][-.a-zA-Z\d]+)$/
-  VERSION_AT_GIT_COMMIT = /\+git\.commit\.([0-9a-f]+)$/
+  VERSION_AT_GIT_COMMIT = /^(\d+[-.][-.a-zA-Z\d]+)\+git\.commit\.([0-9a-f]+)$/
```

Comment:
Since this is using `^`, should it allow (or require) the `v` prefix in the version of the commit?

Shouldn't other tags than `vM.m.p` work? It seems the previous regex allowed them.

Reply:
No, that's a pattern for a version with commit metadata (`1.2.3+git.commit.61e81552724e872e94fb52f95eae02e0e7fa6ffd`)

Tags without the `v` prefix are not allowed. Seems like it was supported once but it's not working now. I'm not sure if I did break it but there are not tests anyway. I'll try to fix it soon.

Reply:
I believe the `v` is required for shard tags. I am missing where the `v` that is in the tag is removed before defining the `package.version.value` to write the lock file.

Reply:
`package.version` is a `Version`, not a tag. One of the intentions with theses changes is having something more meaningful than just strings everywhere.